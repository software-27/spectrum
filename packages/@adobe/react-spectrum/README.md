# @adobe/react-spectrum

This package allows you to easily import any component published under @react-spectrum. Think of it as a helpful alias,
with each component "pinned" to a specific version.

Components are updated sporadically, so if you require a change from a version of a component that is not yet included
in this package, you are able to install the specific version and then update your import statement to use the
component from the @react-spectrum scope. It's worth noting that this may result in increased file size.

## Naming conventions

Any component published under the @react-spectrum scope will be available at the same path from @adobe/react-spectrum.
For example, if you want to use `Checkbox` from @react-spectrum this is what you'd be importing:

```js
import {Checkbox} from '@react-spectrum/checkbox';
```

To import the same component from @adobe/react-spectrum, it would look like this:

```js
import {Checkbox} from '@adode/react-spectrum/checkbox';
```

This convention will be consistent for all packages to make it easy to switch between @react-spectrum and
@adobe/react-spectrum.

## Component versions

| Component                          | Version        |
| ---------------------------------- | -------------- |
| @react-spectrum/actiongroup        | 3.0.0-alpha.1  |
| @react-spectrum/breadcrumbs        | 3.0.0-alpha.1  |
| @react-spectrum/button             | 3.0.0-rc.2     |
| @react-spectrum/buttongroup        | 3.0.0-alpha.1  |
| @react-spectrum/checkbox           | 3.0.0-rc.2     |
| @react-spectrum/dialog             | 3.0.0-alpha.1  |
| @react-spectrum/divider            | 3.0.0-rc.2     |
| @react-spectrum/form               | 3.0.0-rc.2     |
| @react-spectrum/icon               | 3.0.0-rc.2     |
| @react-spectrum/illustratedmessage | 3.0.0-alpha.1  |
| @react-spectrum/image              | 3.0.0-alpha.1  |
| @react-spectrum/label              | 3.0.0-rc.2     |
| @react-spectrum/layout             | 3.0.0-alpha.1  |
| @react-spectrum/link               | 3.0.0-alpha.1  |
| @react-spectrum/listbox            | 3.0.0-alpha.1  |
| @react-spectrum/menu               | 3.0.0-alpha.1  |
| @react-spectrum/meter              | 3.0.0-rc.2     |
| @react-spectrum/overlays           | 3.0.0-alpha.1  |
| @react-spectrum/picker             | 3.0.0-alpha.1  |
| @react-spectrum/progress           | 3.0.0-rc.2     |
| @react-spectrum/provider           | 3.0.0-rc.2     |
| @react-spectrum/radio              | 3.0.0-rc.2     |
| @react-spectrum/searchfield        | 3.0.0-rc.2     |
| @react-spectrum/statuslight        | 3.0.0-rc.2     |
| @react-spectrum/switch             | 3.0.0-rc.2     |
| @react-spectrum/textfield          | 3.0.0-rc.2     |
| @react-spectrum/theme-default      | 3.0.0-rc.2     |
| @react-spectrum/utils              | 3.0.0-rc.2     |
| @react-spectrum/view               | 3.0.0-alpha.1  |
| @react-spectrum/well               | 3.0.0-rc.2     |
