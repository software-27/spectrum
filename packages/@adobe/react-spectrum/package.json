{
  "name": "@adobe/react-spectrum",
  "version": "3.0.0-rc.0",
  "description": "Spectrum UI components in React",
  "license": "Apache-2.0",
  "private": true,
  "main": "dist/main.js",
  "module": "dist/module.js",
  "types": "dist/types.d.ts",
  "source": "src/index.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "scripts": {
    "prepublishOnly": "mkdir -p dist; cp src/index.ts dist/types.d.ts; cp src/index.ts dist/module.js; babel --root-mode upward src/index.ts -o dist/main.js"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/adobe-private/react-spectrum-v3"
  },
  "dependencies": {
    "@react-spectrum/actiongroup": "^3.0.0",
    "@react-spectrum/breadcrumbs": "^3.0.0",
    "@react-spectrum/button": "^3.0.0",
    "@react-spectrum/buttongroup": "^3.0.0",
    "@react-spectrum/checkbox": "^3.0.0",
    "@react-spectrum/dialog": "^3.0.0",
    "@react-spectrum/divider": "^3.0.0",
    "@react-spectrum/form": "^3.0.0",
    "@react-spectrum/icon": "^3.0.0",
    "@react-spectrum/illustratedmessage": "^3.0.0",
    "@react-spectrum/image": "^3.0.0",
    "@react-spectrum/label": "^3.0.0",
    "@react-spectrum/layout": "^3.0.0",
    "@react-spectrum/link": "^3.0.0",
    "@react-spectrum/listbox": "^3.0.0",
    "@react-spectrum/menu": "^3.0.0",
    "@react-spectrum/meter": "^3.0.0",
    "@react-spectrum/overlays": "^3.0.0",
    "@react-spectrum/picker": "^3.0.0",
    "@react-spectrum/progress": "^3.0.0",
    "@react-spectrum/provider": "^3.0.0",
    "@react-spectrum/radio": "^3.0.0",
    "@react-spectrum/searchfield": "^3.0.0",
    "@react-spectrum/statuslight": "^3.0.0",
    "@react-spectrum/switch": "^3.0.0",
    "@react-spectrum/text": "^3.0.0",
    "@react-spectrum/textfield": "^3.0.0",
    "@react-spectrum/theme-dark": "^3.0.0",
    "@react-spectrum/theme-default": "^3.0.0",
    "@react-spectrum/view": "^3.0.0",
    "@react-spectrum/well": "^3.0.0"
  }
}
