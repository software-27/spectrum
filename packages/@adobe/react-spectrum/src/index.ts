/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */

export * from '@react-spectrum/actiongroup';
export * from '@react-spectrum/breadcrumbs';
export * from '@react-spectrum/button';
export * from '@react-spectrum/buttongroup';
export * from '@react-spectrum/checkbox';
export * from '@react-spectrum/dialog';
export * from '@react-spectrum/divider';
export * from '@react-spectrum/form';
export * from '@react-spectrum/icon';
export * from '@react-spectrum/illustratedmessage';
export * from '@react-spectrum/image';
export * from '@react-spectrum/label';
export * from '@react-spectrum/layout';
export * from '@react-spectrum/link';
export * from '@react-spectrum/listbox';
export * from '@react-spectrum/menu';
export * from '@react-spectrum/meter';
export * from '@react-spectrum/overlays';
export * from '@react-spectrum/picker';
export * from '@react-spectrum/progress';
export * from '@react-spectrum/provider';
export * from '@react-spectrum/radio';
export * from '@react-spectrum/searchfield';
export * from '@react-spectrum/statuslight';
export * from '@react-spectrum/switch';
export * from '@react-spectrum/text';
export * from '@react-spectrum/textfield';
export {theme as darkTheme} from '@react-spectrum/theme-dark';
export {theme as defaultTheme} from '@react-spectrum/theme-default';
export * from '@react-spectrum/view';
export * from '@react-spectrum/well';
