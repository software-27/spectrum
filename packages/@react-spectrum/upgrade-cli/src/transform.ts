import {run as jscodeshift} from 'jscodeshift/src/Runner.js';
import path from 'node:path';

const transformPath = path.join(__dirname, 'codemods', 'codemod.js');

interface Options {
  componentSubset?: string,
  path?: string,
  ignorePattern?: string,
  parser?: string
}

export async function transform(options: Options) {
  let {
    path: filePath = '.',
    ...rest
  } = options;

  return await jscodeshift(transformPath, [filePath], rest);
}
