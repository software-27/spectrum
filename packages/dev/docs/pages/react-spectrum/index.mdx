<!-- Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. -->

import {Layout} from '@react-spectrum/docs';
export default Layout;

import {Highlights, Hero} from '@react-spectrum/docs';
import heroNarrow from 'url:../assets/hero-narrow.png';
import heroNarrow2x from 'url:../assets/hero-narrow@2x.png';
import heroWide from 'url:../assets/hero-wide.png';
import heroWide2x from 'url:../assets/hero-wide@2x.png';

# React Spectrum

A React implementation of Spectrum, Adobeâ€™s design system.

<Hero
  narrow={heroNarrow}
  narrow2x={heroNarrow2x}
  wide={heroWide}
  wide2x={heroWide2x}
  alt="React Spectrum hero image" />

<Highlights items={[
  {
    title: 'Accessible',
    description: 'React Spectrum components are designed with accessibility in mind, including full screen reader and keyboard navigation support.',
  },
  {
    title: 'Adaptive',
    description: 'React Spectrum components are designed to work with mouse, touch, and keyboard interactions. Built with responsive design principles to deliver a great experience no matter the device.',
  },
  {
    title: 'Themeable',
    description: 'React Spectrum supports custom themes to match your brand, including automatic support for dark mode.'
  },
  {
    title: 'Internationalized',
    description: 'React Spectrum components are designed to support over 30 languages, including support for right to left, date and number formatting, and more.'
  }
]} />
