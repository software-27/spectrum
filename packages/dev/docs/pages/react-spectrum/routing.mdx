{/* Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {Layout} from '@react-spectrum/docs';
export default Layout;

import {Keyboard} from '@react-spectrum/text';

---
category: Concepts
keywords: [router, links, frameworks, next.js, remix, react-router]
order: 5
---

# Client Side Routing

Many React Spectrum components support rendering as HTML links. This page discusses how to set up your app to integrate React Spectrum links with your framework or client side router.

## Introduction

React Spectrum components such as [Menu](Menu.html), [Tabs](Tabs.html), [TableView](TableView.html), and many others support rendering elements as links that perform navigation when the user interacts with them. Each component that supports link behavior accepts the `href` prop, which causes the component to render an `<a>` element. Other link DOM props such as `target` and `download` are also supported.

By default, links perform native browser navigation when they are interacted with. However, many apps and frameworks use client side routers to avoid a full page reload when navigating between pages. The [Provider](Provider.html) component can configure all React Spectrum components within it to navigate using the client side router you provide. Set this up once in the root of your app, and any React Spectrum component with the `href` prop will automatically navigate using your router.

Note that external links to different origins will not trigger client side routing, and will use native browser navigation. Additionally, if the link has a [target](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target) other than `"_self"`, uses the [download](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#download) attribute, or the user presses modifier keys such as <Keyboard>Command</Keyboard> or <Keyboard>Alt</Keyboard> to change the default behavior, browser native navigation will occur instead of client side routing.

## Provider setup

The `Provider` component accepts a `router` prop, which can be set to a router object. This should include a `navigate` function received from your router for performing a client side navigation programmatically. The following example shows the general pattern. Framework-specific examples are shown below.

```tsx
import {Provider, defaultTheme} from '@adobe/react-spectrum';

function App() {
  let navigate = useNavigateFromYourRouter();

  return (
    <Provider theme={defaultTheme} router={{navigate}}>
      {/* ... */}
    </Provider>
  );
}
```

### React Router

The [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook from `react-router-dom` returns a `navigate` function you can pass to `Provider`. Ensure that the component that calls `useNavigate` and renders `Provider` is inside the router component (e.g. `BrowserRouter`) so that it has access to React Router's internal context. The React Router `<Routes>` element should also be defined inside React Spectrum's `<Provider>` so that links inside the rendered routes have access to the router.

```tsx
import {BrowserRouter, useNavigate} from 'react-router-dom';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

function App() {
  let navigate = useNavigate();

  return (
    <Provider theme={defaultTheme} router={{navigate}}>
      {/* Your app here... */}
      <Routes>
        <Route path="/" element={<HomePage />} />
        {/* ... */}
      </Routes>
    </Provider>
  );
}

<BrowserRouter>
  <App />
</BrowserRouter>
```

### Next.js

#### App router

The [useRouter](https://nextjs.org/docs/app/api-reference/functions/use-router) hook from `next/navigation` returns a router object that can be used to perform navigation. `Provider` should be rendered from a client component at the root of each page or layout that includes React Spectrum components. You can create a new client component for this, or combine it with other top-level providers as described in the [Next.js docs](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#using-context-providers).

```tsx
// app/provider.tsx
"use client";

import {useRouter} from 'next/navigation';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

export function ClientProviders({children}) {
  let router = useRouter();

  return (
    <Provider theme={defaultTheme} router={{navigate: router.push}}>
      {children}
    </Provider>
  );
}
```

Then, in your page or layout server component, wrap your app in the `ClientProviders` component that you defined.

```tsx
// app/layout.tsx
import {ClientProviders} from './provider';

export default function RootLayout({children}) {
  return (
    <html>
      <body>
        <ClientProviders>{children}</ClientProviders>
      </body>
    </html>
  );
}
```

#### Pages router

The [useRouter](https://nextjs.org/docs/pages/api-reference/functions/use-router) hook from `next/router` returns a router object that can be used to perform navigation. `Provider` should be rendered at the root of each page that includes React Spectrum components, or in `pages/_app.tsx` to add it to all pages.

```tsx
// pages/_app.tsx
import type {AppProps} from 'next/app';
import {useRouter} from 'next/router';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

export default function MyApp({Component, pageProps}: AppProps) {
  let router = useRouter();

  return (
    <Provider theme={defaultTheme} router={{navigate: router.push}}>
      <Component {...pageProps} />
    </Provider>
  );
}
```

### Remix

Remix uses React Router under the hood, so the same [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook described above also works in Remix apps. `Provider` should be rendered at the root of each page that includes React Spectrum components, or in `app/root.tsx` to add it to all pages. See the [Remix docs](https://remix.run/docs/en/main/file-conventions/root) for more details.

```tsx
// app/root.tsx
import {useNavigate, Outlet} from '@remix-run/react';
import {Provider, defaultTheme} from '@adobe/react-spectrum';

export default function App() {
  let navigate = useNavigate();

  return (
    <html lang="en">
      <head>
        {/* ... */}
      </head>
      <body>
        <Provider theme={defaultTheme} router={{navigate}}>
          <Outlet />
        </Provider>
        {/* ... */}
      </body>
    </html>
  );
}
```
