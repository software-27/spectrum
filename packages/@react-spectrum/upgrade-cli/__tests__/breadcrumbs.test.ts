// @ts-ignore
import {defineSnapshotTest} from 'jscodeshift/dist/testUtils';
import transform from '../src/codemods/codemod';
jest.mock('../src/getComponents');

const test = (name: string, input: string) => {
  defineSnapshotTest(transform, {}, input, name);
};

test('Static - Renames Item to Breadcrumb and adds import', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';

<div>
  <Breadcrumbs>
    <Item key="home">Home</Item>
    <Item key="trendy">Trendy</Item>
    <Item key="march 2020 assets">March 2020 Assets</Item>
  </Breadcrumbs>
</div>
`);

test('Static - Renames key to id', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';
let externalKey = 'travel';
<div>
  <Breadcrumbs>
    <Item key="news">News</Item>
    <Item key={externalKey}>Travel</Item>
    <Item>Gaming</Item>
    <Item>Shopping</Item>
  </Breadcrumbs>
</div>
`);

test('Dynamic - Renames Item to Breadcrumb', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';
let folders = [
  {id: 1, label: 'Home'},
  {id: 2, label: 'Trendy'},
  {id: 3, label: 'March 2020 Assets'}
];
<div>
  <Breadcrumbs onAction={(a) => setFolderId(a)}>
    {folders.map(f => <Item key={f.id}>{f.label}</Item>)}
  </Breadcrumbs>
</div>
`);

test('Dynamic - Renames key to id', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';
const items = [
  {id: 1, name: 'News'},
  {id: 2, name: 'Travel'},
  {id: 3, name: 'Gaming'},
  {id: 4, name: 'Shopping'}
];
<div>
  <Breadcrumbs items={items}>
    {item => <Item key={item.id}>{item.name}</Item>}
  </Breadcrumbs>
  <Breadcrumbs items={items}>
    {item => <Item id={item.id}>{item.name}</Item>}
  </Breadcrumbs>
  <Breadcrumbs items={items}>
    {item => <Item>{item.name}</Item>}
  </Breadcrumbs>
</div>
`);

test('Comments out showRoot', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';

<Breadcrumbs showRoot>
  <Item key="home">Home</Item>
  <Item key="trendy">Trendy</Item>
  <Item key="2020 assets">March 2020 Assets</Item>
  <Item key="winter">Winter</Item>
  <Item key="holiday">Holiday</Item>
</Breadcrumbs>
`);

test('Comments out autoFocusCurrent', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';

<Breadcrumbs autoFocusCurrent>
  <Item key="home">Home</Item>
  <Item key="trendy">Trendy</Item>
  <Item key="march 2020 assets">March 2020 Assets</Item>
</Breadcrumbs>
`);

test('Comments out isMultiline', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';

<Breadcrumbs isMultiline>
  <Item key="home">Home</Item>
  <Item key="trendy">Trendy</Item>
  <Item key="march 2020 assets">March 2020 Assets</Item>
</Breadcrumbs>
`);

test('Removes size="S"', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';

<Breadcrumbs size="S">
  <Item key="home">Home</Item>
  <Item key="trendy">Trendy</Item>
</Breadcrumbs>
`);

test('Leaves a comment if size prop contains "S"', `
import {Breadcrumbs, Item} from '@adobe/react-spectrum';

<Breadcrumbs size={true ? 'M' : 'S'}>
  <Item key="home">Home</Item>
  <Item key="trendy">Trendy</Item>
</Breadcrumbs>
`);
