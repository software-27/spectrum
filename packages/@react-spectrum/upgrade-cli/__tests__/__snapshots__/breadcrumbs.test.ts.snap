// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments out autoFocusCurrent 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

// TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
// TODO(S2-upgrade): autoFocusCurrent has not been implemented yet.
<Breadcrumbs>
  <Breadcrumb id="home">Home</Breadcrumb>
  <Breadcrumb id="trendy">Trendy</Breadcrumb>
  <Breadcrumb id="march 2020 assets">March 2020 Assets</Breadcrumb>
</Breadcrumbs>"
`;

exports[`Comments out isMultiline 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

// TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
// TODO(S2-upgrade): isMultiline has not been implemented yet.
<Breadcrumbs>
  <Breadcrumb id="home">Home</Breadcrumb>
  <Breadcrumb id="trendy">Trendy</Breadcrumb>
  <Breadcrumb id="march 2020 assets">March 2020 Assets</Breadcrumb>
</Breadcrumbs>"
`;

exports[`Comments out showRoot 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

// TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
// TODO(S2-upgrade): showRoot has not been implemented yet.
<Breadcrumbs>
  <Breadcrumb id="home">Home</Breadcrumb>
  <Breadcrumb id="trendy">Trendy</Breadcrumb>
  <Breadcrumb id="2020 assets">March 2020 Assets</Breadcrumb>
  <Breadcrumb id="winter">Winter</Breadcrumb>
  <Breadcrumb id="holiday">Holiday</Breadcrumb>
</Breadcrumbs>"
`;

exports[`Dynamic - Renames Item to Breadcrumb 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';
let folders = [
  {id: 1, label: 'Home'},
  {id: 2, label: 'Trendy'},
  {id: 3, label: 'March 2020 Assets'}
];
<div>
  // TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
  <Breadcrumbs onAction={(a) => setFolderId(a)}>
    {folders.map(f => <Breadcrumb id={f.id} key={f.id}>{f.label}</Breadcrumb>)}
  </Breadcrumbs>
</div>"
`;

exports[`Dynamic - Renames key to id 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';
const items = [
  {id: 1, name: 'News'},
  {id: 2, name: 'Travel'},
  {id: 3, name: 'Gaming'},
  {id: 4, name: 'Shopping'}
];
<div>
  // TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
  <Breadcrumbs items={items}>
    {item => <Breadcrumb id={item.id}>{item.name}</Breadcrumb>}
  </Breadcrumbs>
  // TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
  <Breadcrumbs items={items}>
    {item => <Breadcrumb id={item.id}>{item.name}</Breadcrumb>}
  </Breadcrumbs>
  // TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
  <Breadcrumbs items={items}>
    {item => <Breadcrumb>{item.name}</Breadcrumb>}
  </Breadcrumbs>
</div>"
`;

exports[`Leaves a comment if size prop contains "S" 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

// TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
<Breadcrumbs // TODO(S2-upgrade): size="S" is no longer supported. You'll need to update this manually.
size={true ? 'M' : 'S'}>
  <Breadcrumb id="home">Home</Breadcrumb>
  <Breadcrumb id="trendy">Trendy</Breadcrumb>
</Breadcrumbs>"
`;

exports[`Removes size="S" 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

// TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
<Breadcrumbs>
  <Breadcrumb id="home">Home</Breadcrumb>
  <Breadcrumb id="trendy">Trendy</Breadcrumb>
</Breadcrumbs>"
`;

exports[`Static - Renames Item to Breadcrumb and adds import 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';

<div>
  // TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
  <Breadcrumbs>
    <Breadcrumb id="home">Home</Breadcrumb>
    <Breadcrumb id="trendy">Trendy</Breadcrumb>
    <Breadcrumb id="march 2020 assets">March 2020 Assets</Breadcrumb>
  </Breadcrumbs>
</div>"
`;

exports[`Static - Renames key to id 1`] = `
"import { Breadcrumb, Breadcrumbs } from "@react-spectrum/s2";
import { Item } from '@adobe/react-spectrum';
let externalKey = 'travel';
<div>
  // TODO(S2-upgrade): S2 Breadcrumbs no longer includes a nav element by default. You can wrap the Breadcrumbs component in a nav element if needed.
  <Breadcrumbs>
    <Breadcrumb id="news">News</Breadcrumb>
    <Breadcrumb id={externalKey}>Travel</Breadcrumb>
    <Breadcrumb>Gaming</Breadcrumb>
    <Breadcrumb>Shopping</Breadcrumb>
  </Breadcrumbs>
</div>"
`;
